dns/README.md

```diff
@@ -0,0 +1,107 @@
+| Name         | 16 bits        | \xc0\x0c                                                  |
+| Type         | 16 bits        | value is 12                                               |
+| Class        | 16 bits        | value is 1 (Internet: IN)                                 |
+| TTL          | 32 bits        | unsigned integer that specifies the time period in seconds|
+| Data length  | 16 bits        | the length of Data field (unit: byte)                     |
+| Data         | variable size  | host name                                                 |
+
+<br>
+
+### SRV RR
+| Field        | Size           | Note                                                       |
+| ------------ |:--------------:| -----:                                                     |
+| Name         | 16 bits        | \xc0\x0c                                                   |
+| Type         | 16 bits        | value is 33                                                |
+| Class        | 16 bits        | value is 1 (Internet: IN)                                  |
+| TTL          | 32 bits        | unsigned integer that specifies the time period in seconds |
+| Data length  | 16 bits        | the length of Data field (unit: byte)                      |
+| Data         | variable size  | Data field consists of Priority, Weight, Port, and Target  |
+| Priority     | 16 bits        | unsigned integer                                           |
+| Weight       | 16 bits        | unsigned integer                                           |
+| Port         | 16 bits        | range from 0 -65535                                        |
+| Target       | variable size  | target's host name                                         |
+
+<br>
+
+### What about multiple answers?
+You will need to ANCOUNT to the number of answers in the header section in the resposne packets, and combine all the answers in the answer section.
+
+For example, if you are returning two answers, first, set the ANCOUNT to 2. And pack two anwsers back to back like this:
+
+Name 1 | Type 1 | Class 1| TTL 1| Data length 1| Data 1| Name 2 | Type 2 | Class 2| TTL 2| Data length 2| Data 2
+
+## Why we need DNS server and SRV records in piman?
+In order to use Prometheus on piman and export the metric data from all the pis, we need to specify where to gather data, and in this case, it will be all the pi's IP addresses and port 9100 if we use node_exporter.
+
+However, there will be **prometheus.yml** that specifes where to gather data and **hosts.csv** that contains all the pis' IPs, and if we need to add one more pi, we will need to update two files. Hence, to keep it simple and easy to maintain, we use SRV records to indicate the services' location, which means that we only need to specify target as **metrics.city.cs158b** in **prometheus.yml**, and the DNS server (piman) will resolve all the pi's addresses and ports for us (node_exporter).
```

Comment:
city.cs158b is just for usein class. you can keep city.cs158b but make it clear that it is an example name.